### ISSUE TITLE
Chuck as a puredata or max/msp external?

### ISSUE THREAD

**Issue by shakfu** [ccrma/chuck#203]
I have noted that in the recent version (1.4.1.*) of chuck there is this separation of the core code from the host code and that one can generate a `libchuck.a` with the current Xcode project in the chuck source, and that all of this refactoring is aimed at making embedding chuck easier.

However, I am struggling to find simple examples of embedding chuck, or documentation of the chuck cpp or c api so that I can figure out whether it makes sense to embed chuck in a puredata or max/msp external in the same way that csound is embedded for example or faust is embedded for example.

S

**Comment by shakfu** [comment]
FYI, I've made some decent progress on the above recently, and just about got a puredata wrapper (based on to numchucks codebase) to compile without errors. It still not functional yet or sound-producing, but seems like a promising start. Ongoing work on this is in this [repo](https://github.com/shakfu/pd-chuck).

Since I've managed to integrate so far without including `RtAudio`, which did seem like a hard dependency, I'm assuming that it should be workable for Chuck to just integrate with PD's sound generation mechanisms.

Any feedback, or help, of course, would be much appreciated.
